### PR TITLE
Cache recourses between tests

### PR DESCRIPTION
**Pull request by miherlosev**
Changes:
* add the `--cache` option that turn on the caching resources (script and styles) between test runs.
* release the `http.Server` in tests (https://github.com/nodejs/node/issues/2642) correctly.

### CODE REVIEW

File: src/cli/argument-parser.ts [hunk 1]
```diff
@@ -166,7 +166,8 @@ export default class CLIArgumentParser {
         this.experimental
             .allowUnknownOption()
             .option('--disable-multiple-windows', 'disable multiple windows mode')
-            .option('--experimental-compiler-service', 'run compiler in a separate process');
+            .option('--experimental-compiler-service', 'run compiler in a separate process')
+            .option('--cache', 'cache resources between test runs');
```
Comment (DIRECTcut) [inline src/cli/argument-parser.ts:170]:
```suggestion
            .option('--preserve-cache', 'preserve browser cache between test runs');
```

here and below

Comment (miherlosev) [reply src/cli/argument-parser.ts:170]: In some tools, the `cache` option is used:
- GitLab (https://docs.gitlab.com/ee/ci/caching/)
- Jest (https://jestjs.io/docs/en/cli)
- npm (https://docs.npmjs.com/cli/v6/commands/npm-cache)

Comment (DIRECTcut) [reply src/cli/argument-parser.ts:170]: > TestCafe doesn't use the browser cache. TestCafe has own cache and store some web application resources (script and styles) between test runs.

That makes a difference. `--cache` is good then

```suggestion
            .option('--cache', 'cache web assets between test runs');
```

Comment (miherlosev) [reply src/cli/argument-parser.ts:170]: Ok.

**Comment by miherlosev** [comment]
FPR. The tests are green. The TestCafe build bot didn't add a necessary comment.

**Review by AndreyBelym — APPROVED** [review]

**Pull request merged**